File: examples/heartbeat-client.py
#!/usr/bin/env python3
"""
Cloudflare Worker Heartbeat Client (Python)
This script sends a heartbeat to the Cloudflare Worker
Run this periodically using cron, systemd timer, or a scheduler
"""

import requests
import socket
import json
import sys
from datetime import datetime

# Configuration
WORKER_URL = "https://heartbeat-monitor.your-subdomain.workers.dev/api/heartbeat"
SERVICE_ID = "service-1"
API_KEY = "your-secret-key-1"

def send_heartbeat():
    """Send heartbeat to Cloudflare Worker"""
    
    # Gather metadata
    hostname = socket.gethostname()
    timestamp = datetime.utcnow().isoformat() + "Z"
    
    # Build payload
    payload = {
        "serviceId": SERVICE_ID,
        "status": "up",
        "metadata": {
            "hostname": hostname,
            "timestamp": timestamp,
        },
        "message": f"Heartbeat from {hostname}"
    }
    
    headers = {
        "Content-Type": "application/json",
        "Authorization": f"Bearer {API_KEY}"
    }
    
    try:
        # Send heartbeat
        response = requests.post(
            WORKER_URL,
            json=payload,
            headers=headers,
            timeout=10
        )
        
        # Check response
        if response.status_code == 200:
            result = response.json()
            print(f"[{datetime.now()}] Heartbeat sent successfully: {result.get('message')}")
            return 0
        else:
            print(f"[{datetime.now()}] Failed to send heartbeat: {response.status_code} - {response.text}")
            return 1
            
    except requests.exceptions.RequestException as e:
        print(f"[{datetime.now()}] Error sending heartbeat: {e}")
        return 1

if __name__ == "__main__":
    sys.exit(send_heartbeat())

File: examples/heartbeat-client.js
#!/usr/bin/env node

/**
 * Cloudflare Worker Heartbeat Client (Node.js)
 * This script sends a heartbeat to the Cloudflare Worker
 * Run this periodically using cron, systemd timer, or a scheduler
 */

const https = require('https');
const os = require('os');

// Configuration
const WORKER_URL = 'https://heartbeat-monitor.your-subdomain.workers.dev/api/heartbeat';
const SERVICE_ID = 'service-1';
const API_KEY = 'your-secret-key-1';

/**
 * Send heartbeat to Cloudflare Worker
 */
async function sendHeartbeat() {
  // Gather metadata
  const hostname = os.hostname();
  const timestamp = new Date().toISOString();
  
  // Build payload
  const payload = JSON.stringify({
    serviceId: SERVICE_ID,
    status: 'up',
    metadata: {
      hostname: hostname,
      timestamp: timestamp,
      platform: os.platform(),
      arch: os.arch(),
    },
    message: `Heartbeat from ${hostname}`
  });
  
  // Parse URL
  const url = new URL(WORKER_URL);
  
  const options = {
    hostname: url.hostname,
    port: 443,
    path: url.pathname,
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
      'Content-Length': Buffer.byteLength(payload),
      'Authorization': `Bearer ${API_KEY}`
    }
  };
  
  return new Promise((resolve, reject) => {
    const req = https.request(options, (res) => {
      let data = '';
      
      res.on('data', (chunk) => {
        data += chunk;
      });
      
      res.on('end', () => {
        if (res.statusCode === 200) {
          console.log(`[${new Date().toISOString()}] Heartbeat sent successfully:`, data);
          resolve(0);
        } else {
          console.error(`[${new Date().toISOString()}] Failed to send heartbeat: ${res.statusCode} - ${data}`);
          resolve(1);
        }
      });
    });
    
    req.on('error', (error) => {
      console.error(`[${new Date().toISOString()}] Error sending heartbeat:`, error.message);
      resolve(1);
    });
    
    req.write(payload);
    req.end();
  });
}

// Run
sendHeartbeat().then(code => process.exit(code));

